spec/services/user_state_transition_segment_service_spec.rb

```diff
@@ -111,20 +126,26 @@
       end
 
       it 'calls SegmentService#identify with proper arguments' do
-        allow_any_instance_of(SegmentService).to receive(:track).with(
-          'user_insufficient'
-        )
-        expect_any_instance_of(SegmentService).to receive(:identify).with(
+        expect(magic_segment_service).to receive(:identify).with(
           state: 'insufficient'
         )
+        expect(magic_segment_service).to receive(:identify).with(
+          pull_requests_count: 4
```

Comment:
This may be a more general question about our use of the `insufficient` event: in terms of how we would be using it in Segment, would it make sense to think about testing for the more normative case (fewer than 4 PRs) here?

Reply:
This spec seems to be testing the Segment service directly without actually using the user model correctly. A user with 4 PRs wouldn't actually be able to transition to the insufficient state.

This might be a flaw in this segment logic actually, as the user has no insufficient state, only the insufficient event which moves the back to being registered.

Reply:
Ah yeah – like, the transition itself is meant to function as a kind of representation of the user's state, but the kind of logical flow that's possible in the user model makes what's being sent to Segment ... marginally meaningful? At least in this case.

Reply:
Actually, I don't think that matters, the state in Segment can be different from the state the user has in the DB. I'll just update this spec to stub the user with an insufficient number of PRs, to the transition makes sense in context.

Reply:
Sounds good!

Reply:
I ended up just removing the tests for the pull requests count here, as it was getting very confusing as to where they were firing. There is a dedicated spec for testing just the PR count updates to Segment 👍